### PR TITLE
code clean up

### PR DESCRIPTION
**Pull request by chrisgeek**
Code Clean up
Search form works perfectly now
The search filter on the left side of the page needs to be looked at and fixed in another PR, the resource variable is no longer available. 
Need the reviewers to test the job refactoring and ensure it works as expected.

### CODE REVIEW

File: app/controllers/jobs_controller.rb
Comment (daniel-sussman) [inline app/controllers/jobs_controller.rb:1]: Great job on the search implementation. All of the if params[:resource].present? &c. stuff can be replaced by your lines 73-77. However, all code above line 33 is necessary for the sidebar logic. This file seems to revert that code to an earlier version, before I revised it in response to changes to the view files.

File: app/models/job.rb
Comment (daniel-sussman) [inline app/models/job.rb:1]: if geo = results.first does two things at once: 1) checks whether the geomapping API was able to associate the job's location string with any results, and 2) it sets geo to the first (i.e. most probable) geospatial result. Double equals sign is not intended.

File: app/views/jobs/_filter_category_sidebar.html.erb
Comment (daniel-sussman) [inline app/views/jobs/_filter_category_sidebar.html.erb:1]: The changes to resources will break the sidebar. Please revert them for now.

File: app/views/jobs/index.html.erb
Comment (daniel-sussman) [inline app/views/jobs/index.html.erb:1]: Please don't comment out the filter sidebar!

File: app/services/job_application/apply_job_service.rb
Comment (daniel-sussman) [inline app/services/job_application/apply_job_service.rb:1]: Ilya can confirm this, but pretty sure DefaultMale and DefaultFemale can be removed.

**Review by daniel-sussman — CHANGES_REQUESTED** [review]
Nice work, Chris. Let's discuss how best to refactor the sidebar logic.

File: app/services/job_application/apply_job_service.rb
Comment (Ches-ctrl) [reply app/services/job_application/apply_job_service.rb:1]: Let's hold on removing these as I'd like there to be default values for users

**Pull request closed without merging**